aten/src/ATen/native/cpu/UpSampleKernelAVXAntialias.h

```diff
@@ -534,22 +656,76 @@ void ImagingResampleHorizontalConvolution8u4x(
 
     // Write the output into single uint32
     // (a b c d) -> x_uint32
-    lineOut0[out_x] = _mm_cvtsi128_si32(_mm256_extracti128_si256(sss0, 0));
-    lineOut1[out_x] = _mm_cvtsi128_si32(_mm256_extracti128_si256(sss0, 1));
-    lineOut2[out_x] = _mm_cvtsi128_si32(_mm256_extracti128_si256(sss1, 0));
-    lineOut3[out_x] = _mm_cvtsi128_si32(_mm256_extracti128_si256(sss1, 1));
+    auto o0 = _mm_cvtsi128_si32(_mm256_castsi256_si128(sss0));
+    auto o1 = _mm_cvtsi128_si32(_mm256_extracti128_si256(sss0, 1));
+    auto o2 = _mm_cvtsi128_si32(_mm256_castsi256_si128(sss1));
+    auto o3 = _mm_cvtsi128_si32(_mm256_extracti128_si256(sss1, 1));
+
+    const auto out_x_strided = stride * out_x;
+
+    if (num_channels == 3 && C10_UNLIKELY(out_x_strided + 4 >= max_out_x_strided)) {
+      // This is a boundary case when we want to write 4 bytes to the output buffer but
+      // the 4th bytes is already computed. It means that we can not overwrite it.
```

Comment:
I'm not sure I get why we can't ovewrite a byte because it has already been computed. Could you help me understand?

Reply:
I think I have to add here that forth byte in the register that we want to write the output has a "garbage" value for num_channels = 3. So, if we write it to the output we'll overwrite good value with a wrong value like in the diagram below.